File: tamer/cp.cpp
#include <iostream>
#include <fstream>

/*************************************************************************************************/
void cp_by_line(std::istream& src, std::ostream& dst) {
    std::string line;
    
    while (std::getline(src, line)) {
        dst << line << std::endl;
    }
}



/*************************************************************************************************/
int main(int argc, char* argv[]) {
    if (argc > 2) {
        std::ifstream src(argv[1]);
        std::ofstream dst(argv[2]);

        if (src.is_open() && dst.is_open()) {
            cp_by_line(src, dst);
        }

        src.close();
        dst.close();
    }

    return 0;
}

File: village/sketch/CP.java
import java.lang.Exception;

import java.util.Scanner;

import java.io.InputStream;
import java.io.FileInputStream;
import java.io.OutputStream;
import java.io.FileOutputStream;

public class CP { 
    public static void cp_by_line(InputStream src, OutputStream dst) {
        try {
            Scanner s = new Scanner(src);

            while (s.hasNextLine()) {
                String line = s.nextLine();

                dst.write(line.getBytes(), 0, line.length());
                dst.write('\n');
            }

            s.close();
        } catch (Exception e) {}
    }

    public static void main(String argv[]) {
        if (argv.length > 0) {
            try {
                FileInputStream src = new FileInputStream(argv[0]);

                if (argv.length == 1) {
                    cp_by_line(src, System.out);
                } else {
                    try {
                        FileOutputStream dst = new FileOutputStream(argv[1]);
                        
                        cp_by_line(src, dst);
                        dst.close();
                    } catch (Exception e) {
                        System.out.println(e.getMessage());
                    }
                }

                src.close();
            } catch (Exception e) {
                System.out.println(e.getMessage());
            }
        } else {
            cp_by_line(System.in, System.out);
        }
    }
}
